### DISCUSSION TITLE
Any reason why LangChain.Message.new_tool_result/1 only supports content field?

### DISCUSSION THREAD

**Discussion by CaiqueMitsuoka** [brainlid/langchain#219] [General]
Asking about [LangChain.Message.new_tool_result/1](https://github.com/brainlid/langchain/blob/a832d05b52ba9ad66bc99b5a8220a4f994c4b2c2/lib/message.ex#L384) that only populates `content` and `tool_calls`(cursively).
- Is there any reason why it doesn't support other fields line `tool_call_id` and `name`?

Just to give context on my use case:
I save the exchanged messages(user and assistant) and tool_calls(name, args, result) associated with them.
Across requests I need reconstruct the chain and use result from previous tools. But I need to validate which tool was called.

Quick example:

**User**: Send an email to Brad asking if he is coming to the party.
**Tool** `create_email` is called
**Assistant**: Nice, I found Brad Pit on your contact list. Here is the email:

Hey Brad, are you coming to the party tomorrow?

Can I send the email?
**User**: Yep
**Tool** `send_email`, validates that the request is an `create_email` and it uses the output execute to the action.

I know its a silly example, but for compliance this is very important requirement. Once the user agree with something we need to stick with it.
The field name would allow to know the type of the on callbacks without a database trip to parse the result according to the tool format. The field `tool_call_id` allow to more complex queries to be done when only the name of the tool and the result is not enough, in this example with the `tool_call_id` I can find in the database the call and find the contact id that was not included in the tool result.

I'm able to achieve all of this with
```elixir
%{}
|> Map.put(:content, result)
|> LangChain.Message.new_tool_result!()
|> Map.put(:tool_call_id, id)
|> Map.put(:name, name)
```

But since the api still in rc I figure this is a quick and easy win.
I can open a PR if it's ok 👍

**Comment by brainlid** [comment]
Hi @CaiqueMitsuoka!

I may be missing something here, but the `ToolResult` does include `tool_call_id` and `name`. Unless there's a disconnect with what you're needing.

https://github.com/brainlid/langchain/blob/main/lib/message/tool_result.ex#L26

You can do this:

```elixir
Message.new_tool_result!(%{
  tool_results: [
    ToolResult.new!(%{
      tool_call_id: "call_123",
      name: "hello_world",
      content: "Hello world!"
    })
  ]
})
```

Does this meet your needs?

**Reply by brainlid** [reply]
And these responses are auto-created when a `LangChain.Function` is executed and returned. Those created results are added to the chain. If needed, you could port those over to your new chain. Though, I'm not really clear on your use-case.
